lib/kitchen/patches/nokogiri.rb

```diff
@@ -74,6 +74,14 @@ def previous
         prev.text? ? prev.previous : prev
       end
 
+      def preceded_by_text?
+        prev = previous_sibling
+        while !prev.nil? && prev.blank? do prev = prev.previous_sibling end
+        return false if prev.nil?
+
+        prev.text?
+      end
```

Comment:
Why is this using `previous_sibling` whereas `previous` above is using `previous_element`?  

I'm also confused now why the `previous` implementation even checks for `text?` because per the [Nokogiri docs](https://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/Node#previous_element-instance_method), `previous_element` only returns element node types, i.e. not text nodes???

Reply:
I'm not entirely sure--- it's using previous_sibling to pick up text-- as I said in the previous comment, apparently previous_element on paper does exactly what we want it to do-- (instead of the  nokigiri #previous/#previous sibling which just returns the previous node) perhaps we could get rid of the #previous patch, and switch Node_Element#previous to use raw.previous_element?